### PR TITLE
Reconcile differences from upstream finishApplyDynamicCallable.

### PR DESCRIPTION
**Pull request by pschuh**
There are still some differences, but this is a pre-step to resolving those.

### CODE REVIEW

File: lib/Sema/CSApply.cpp [hunk 1]
```diff
@@ -6903,92 +6906,76 @@ static Expr *finishApplyDynamicCallable(ExprRewriter &rewriter,
...
-                                          TupleTypeElt{valueAssocType}},
-                                         ctx));
-      dictElements.push_back(dictElt);
-    }
-    auto *dictExpr = DictionaryExpr::create(
-        ctx, arg->getStartLoc(), dictElements, {}, arg->getEndLoc());
-    cs.setType(dictExpr, expectedParamType);
-    rewriter.finishDictionaryExpr(dictExpr);
-    argument = dictExpr;
+      Expr *labelExpr =
```
Comment (dan-zheng) [inline lib/Sema/CSApply.cpp]: Consider consistent 4-space indentation in this function body.

**Review by dan-zheng — APPROVED** [review]
LGTM if tests pass!

I believe this at least partially resolves [TF-217](https://bugs.swift.org/browse/TF-217). I tried to upstream to `master` some `@dynamicCallable`-fixed-crasher tests, but didn't create `Python`-independent reproducers. Consider closing the issue if it makes sense to you. 🙂

File: lib/Sema/CSApply.cpp (hunk 1 above)
Comment (dan-zheng) [reply lib/Sema/CSApply.cpp]: Never mind - this seems to be exactly in sync with [`master`](https://github.com/apple/swift/blob/10a4e3ef41059608ffe9eded039350055064ce9f/lib/Sema/CSApply.cpp#L6814).

**Comment by pschuh** [comment]
This is not exactly in sync. What is missing from master is the coercion logic.

**Comment by pschuh** [comment]
@swift-ci please test tensorflow

**Comment by pschuh** [comment]
@swift-ci please test tensorflow

**Comment by pschuh** [comment]
@swift-ci please test tensorflow

**Comment by pschuh** [comment]
Should now be perfectly in sync.

**Comment by pschuh** [comment]
@swift-ci please test tensorflow

**Pull request merged**